src/workers/postUpdated.ts

```diff
@@ -0,0 +1,120 @@
+import { messageToJson, Worker } from './worker';
+import * as he from 'he';
+import {
+  addTagsAndKeywords,
+  ArticlePost,
+  bannedAuthors,
+  findAuthor,
+  mergeKeywords,
+  parseReadTime,
+  PostOrigin,
+  SharePost,
+  Source,
+} from '../entity';
+
+interface Data extends Omit<ArticlePost, 'keywords' | 'tags'> {
+  updated_at: Date;
+  keywords: string[];
+  tags: string[];
+}
+
+const worker: Worker = {
+  subscription: 'api.content-published',
+  handler: async (message, con, logger): Promise<void> => {
+    const data: Data = messageToJson(message);
+    try {
+      const { id, tags, keywords, updated_at, ...submittedData } = data;
+      const updatedDate = new Date(updated_at);
+      await con.transaction(async (entityManager) => {
+        // For now, we only allow Squad posts to be updated through this flow
+        const databasePost = await entityManager
+          .getRepository(ArticlePost)
+          .findOneBy({ id, origin: PostOrigin.Squad });
+
+        if (
+          !databasePost ||
+          databasePost.metadataChangedAt.toISOString() >=
+            updatedDate.toISOString()
+        ) {
+          return;
+        }
+
+        if (bannedAuthors.indexOf(submittedData?.creatorTwitter) > -1) {
+          logger.info(
+            { data, messageId: message.messageId },
+            'post update failed because author is banned',
+          );
+          return;
+        }
+
+        const creatorTwitter =
+          submittedData.creatorTwitter === '' ||
+          submittedData.creatorTwitter === '@'
+            ? null
+            : submittedData.creatorTwitter;
+
+        const authorId = await findAuthor(entityManager, creatorTwitter);
+        const becomesVisible =
+          !databasePost?.visible && !!submittedData?.title?.length;
+
+        const { allowedKeywords, mergedKeywords } = await mergeKeywords(
+          entityManager,
+          keywords,
+        );
+
+        if (allowedKeywords.length > 5) {
+          logger.info(
+            {
+              url: submittedData.url,
+              keywords: allowedKeywords,
+            },
+            'created an article with more than 5 keywords',
+          );
+        }
+
+        const { private: privacy } = await entityManager
+          .getRepository(Source)
+          .findOneBy({ id: submittedData?.sourceId });
+
+        const fixedData = {
+          ...submittedData,
+          authorId,
+          creatorTwitter,
+          canonicalUrl: data?.canonicalUrl || data?.url,
+          title: data?.title && he.decode(data?.title),
+          readTime: parseReadTime(data?.readTime),
+          publishedAt: data?.publishedAt && new Date(data?.publishedAt),
+          metadataChangedAt: updatedDate,
+          visible: becomesVisible,
+          visibleAt: becomesVisible ? updatedDate : null,
+          tagsStr: allowedKeywords?.join(',') || null,
+          private: privacy,
+          sentAnalyticsReport: !data?.authorId,
```

Comment:
If the post is private we shouldn't send analytics report

```suggestion
          sentAnalyticsReport: private || !data?.authorId,
```